documentation/api/1.0/api.md

```diff
@@ -0,0 +1,864 @@
+	// convert the username/groupnames to UID/GID values.
+	//
+	// Note that this argument applies solely to the _runtime_ supporting
+	// a "--user=" option when running in a "standalone mode" - there is
+	// no issue when the runtime is called by a container manager since
+	// all the user and group mapping is handled by the container manager
+	// and specified to the runtime in terms of UID/GID's in the
+	// configuration file generated by the container manager.
+	User                string
+	PrimaryGroup        string
+	SupplementaryGroups []string
+
+	Interactive     bool
+	Console         string
+	Detach          bool
+	NoNewPrivileges bool
+	Capabilities    LinuxCapabilities
+}
+```
+
+###### `Mount`
+```Go
+// Mount describes a container mount.
+type Mount struct {
+	Source      string
+	Destination string
+
+	// Type specifies the type of filesystem to mount.
+	Type string
+
+	// Options list all the mount options of the filesystem.
+	Options []string
+
+	// HostPath used to store host side bind mount path
+	HostPath string
+
+	// ReadOnly specifies if the mount should be read only or not
+	ReadOnly bool
+}
+```
+
+###### `DeviceInfo`
+```Go
+// DeviceInfo is an embedded type that contains device data common to all types of devices.
+type DeviceInfo struct {
+	// Device path on host
+	HostPath string
+
+	// Device path inside the container
+	ContainerPath string
+
+	// Type of device: c, b, u or p
+	// c , u - character(unbuffered)
+	// p - FIFO
+	// b - block(buffered) special file
+	// More info in mknod(1).
+	DevType string
+
+	// Major, minor numbers for device.
+	Major int64
+	Minor int64
+
+	// FileMode permission bits for the device.
+	FileMode os.FileMode
+
+	// id of the device owner.
+	UID uint32
+
+	// id of the device group.
+	GID uint32
+
+	// Hotplugged is used to store device state indicating if the
+	// device was hotplugged.
+	Hotplugged bool
+
+	// ID for the device that is passed to the hypervisor.
+	ID string
+}
+```
+
+#### `VCPod`
+```Go
+// VCPod is the Pod interface
+// (required since virtcontainers.Pod only contains private fields)
+type VCPod interface {
+	Annotations(key string) (string, error)
+	GetAllContainers() []VCContainer
+	GetAnnotations() map[string]string
+	GetContainer(containerID string) VCContainer
+	ID() string
+	SetAnnotations(annotations map[string]string) error
+}
+```
+
+### Pod Functions
+
+* [CreatePod](#createpod)
+* [DeletePod](#deletepod)
+* [StartPod](#startpod)
+* [StopPod](#stoppod)
+* [RunPod](#runpod)
+* [ListPod](#listpod)
+* [StatusPod](#statuspod)
+* [PausePod](#pausepod)
+* [ResumePod](#resumepod)
+
+#### `CreatePod`
+```Go
+// CreatePod is the virtcontainers pod creation entry point.
+// CreatePod creates a pod and its containers. It does not start them.
+func CreatePod(podConfig PodConfig) (VCPod, error)
+```
+
+#### `DeletePod`
+```Go
+// DeletePod is the virtcontainers pod deletion entry point.
+// DeletePod will stop an already running container and then delete it.
```

Comment:
Not directly related to this PR probably, but it would be nice if the func docs here noted when you should and should not call the funcs (like `not to be called on a running container` or similar, and what you can expect to get back in the cases of success and failure.